plugin/distribute.go

```diff
@@ -0,0 +1,165 @@
+package plugin
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/bytedance/sonic"
+	"github.com/redis/go-redis/v9"
+	"github.com/vermaShivansh/coraza-ratelimit-plugin/helpers"
+)
+
+type Distributed struct {
+	Active       bool          // if ratelimit has to be distributed
+	SyncInterval time.Duration // syncing with redis at interval
+	lastSync     int64         // timestamp for lastSync with redis
+	UniqueKey    string        // has to be same for all the instances which have to be synced together
+}
+
+var RATELIMIT_KEY = "coraza_ratelimit_key"
+
+// initiate distribution
+func (e *Ratelimit) initDistribute(syncInterval time.Duration) error {
+	var err error
+
+	unique_key := os.Getenv(RATELIMIT_KEY)
+
+	if unique_key == "" {
+		return errors.New("please set a env value with keyname `coraza_ratelimit_key` to use ratelimit distribution ")
+	}
+
+	if err = helpers.CheckRatelimitDistributeKey(unique_key); err != nil {
```

Comment:
I would not check this input. I would just make sure it isn't empty.

Reply:
but won't this cause cases where multiple users can just type a single letter suppose 'a' or something like 'my-application' (basically common names) which will clash with other instances which shouldn't be synced.
I kept the checks which promote the uniqueness of keys. 
What do you think, Should we not do this on our end or How will we prevent clashes among instances?

Reply:
I would suggest you append a small hash (6 chars like a commit), something like docker containers do. In general validating user input is problematic and should be avoided.

Reply:
but if I append a small hash, then it has to be same for all right?
it will be a fixed hash because if we generate it everytime it will even differ for same unique_keys and instances couldn't be synced.

Reply:
okay, thanks for the advice. 😸 🙇